core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/migration/WorkspaceMigrationActor.scala

```diff
@@ -402,13 +410,148 @@ object WorkspaceMigrationMonitor {
+        (env.storageTransferService, GoogleProject(env.billingProject.googleProjectId.value))
+      }
+
+      outcome <- MigrateAction.liftF {
+        OptionT {
+          storageTransferService
+            .listTransferOperations(record.jobName, billingProject)
+            .map(_.foldLeft(Option.empty[Outcome]) {
+              (outcome, operation) => outcome |+| getOperationOutcome(operation)
+            })
+        }
+      }
+
+      (status, message) = toTuple(outcome)
+      _ <- inTransaction { _ =>
+        storageTransferJobs
+          .filter(_.id === record.id)
+          .take(1)
+            .map(row => (row.outcome, row.message))
+            .update(status, message)
+      }
+    } yield (record.migrationId, outcome)
+
+
+  final def getOperationOutcome(operation: Operation): Option[Outcome] =
+    operation.getDone.some.collect { case true =>
+      if (!operation.hasError)
+        Success.asInstanceOf[Outcome]
+      else {
+        val status = operation.getError
+        if (status.getDetailsCount == 0)
+          Failure(status.getMessage) else
+          Failure(status.getMessage ++ " : " ++ status.getDetailsList.toString)
+      }
+    }
+
+
+  final def transferJobSucceeded(migrationId: Long): MigrateAction[Unit] =
+    nowTimestamp.flatMap { transferred =>
+      inTransaction { dataAccess =>
+        val migrations = workspaceMigrations.filter(_.id === migrationId)
+        // we can encode which bucket was transferred by filtering instead of explicitly looking
+        // it up and branching. If the workspace bucket was already transferred then the job
+        // relates to the tmp bucket, otherwise the job relates to the workspace bucket.
+        DBIO.seq(
+          migrations
+            .filter(m => m.workspaceBucketTransferred.isDefined && m.tmpBucketTransferred.isDefined)
+            .take(1)
+            .result
+            .headOption
+            .flatMap {
+              case None => DBIO.successful()
+              // If this triggers then a job has been issued more than once. This is likely caused
+              // by running the migration pipeline in parallel.
+              case Some(m) => dataAccess
+                .workspaceQuery
+                .findByIdOrFail(m.workspaceId.toString)
+                .flatMap { workspace =>
+                  DBIO.failed(new RawlsException(
+                    s"""Workspace migration "$migrationId" has already transferred
+                       |workspace "${m.workspaceId}" bucket "${workspace.bucketName}"
+                       |to Google Project "${m.newGoogleProjectId}".""".stripMargin
+                  ))
+                }
+            },
+          migrations
+            .filter(_.workspaceBucketTransferred.isDefined)
+            .map(_.tmpBucketTransferred)
+            .update(transferred.some),
+          migrations
+            .filter(_.workspaceBucketTransferred.isEmpty)
+            .map(_.workspaceBucketTransferred)
```

Comment:
I noticed this has a different format than above, but is still approaching the same type of operation (bucket transferred). Is there a reason to not use the same `isDefined` or `isEmpty` for both of them? E.g.,
```
migrations
  .filter(_.workspaceBucketTransferred.isDefined)
  .map(_.tmpBucketTransferred)
  .update(transferred.some),
migrations
  .filter(_.finalBucketTransferred.isDefined) //Whatever this variable should be
  .map(_.workspaceBucketTransferred)
  .update(transferred.some)
```

Because they are so similar as it is, but the implementation is different, I am worried that future us is going to do the same thing I just did, and assume they are checking in the same way, when they arent.

Reply:
If the temp bucket has been transferred already then that's an error. What I'm trying to do is split the migrations by if they've had their workspace bucket transferred to the tmp bucket. To those that have, I'm setting the flag that the tmp bucket has been transferred. To those that haven't I'm setting it.

Reply:
I've added a comment to clarify